src/App.tsx

```diff
@@ -1,61 +1,148 @@
+import React, { useState, FormEvent } from 'react';
+import cn from 'classnames';
+
+import 'bulma/css/bulma.min.css';
 import './App.scss';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList/TodoList';
+import { Todo } from './types/Todo';
+import { User } from './types/User';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+
+function getUser(userId: number): User | null {
+  const foundUser = usersFromServer.find(user => user.id === userId);
+
+  return foundUser || null;
+}
+
+const preparedTodos: Todo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+const getMaxId = (todos: Todo[]) => Math.max(...todos.map(todo => todo.id));
 
 export const App = () => {
+  const [todos, setTodos] = useState(preparedTodos);
+  const [title, setTitle] = useState('');
+  const [user, setUser] = useState(-1);
+  const [completed, setCompleted] = useState(false);
+  const [isInvalidTitle, setTitleCondition] = useState(false);
+  const [isInvalidUser, setUserCondition] = useState(false);
+
+  const handleTitle = (action: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(action.target.value.replace(/[^a-z0-9а-я ]/gi, ''));
+    setTitleCondition(false);
+  };
+
+  const handleUser = (action: React.ChangeEvent<HTMLSelectElement>) => {
+    setUser(+action.target.value);
+    setUserCondition(false);
+  };
+
+  const handleCompleted = (action: React.ChangeEvent<HTMLInputElement>) => {
+    setCompleted(action.target.checked);
+  };
+
+  const makeNewTodo = (action: FormEvent) => {
+    action.preventDefault();
+
+    setTitleCondition(title.length === 0);
+    setUserCondition(user === -1);
+
+    if (title.length !== 0 && user !== -1) {
+      setTodos([
+        {
+          id: getMaxId(todos) + 1,
+          title,
+          completed,
+          userId: usersFromServer[user - 1].id,
+          user: usersFromServer[user - 1],
+        },
+        ...todos,
+      ]);
```

Comment:
For the test, you should add a new todo to the end of todos, but I think it's not a big deal in this case
1) should add a todo to the end of the list